types/types.go

```diff
@@ -15,5 +16,10 @@ type WarpLogInfo struct {
 	SourceTxID       []byte
 	UnsignedMsgBytes []byte
 	BlockNumber      uint64
-	IsCatchUpMessage bool
+}
+
+type WarpBlockInfo struct {
```

Comment:
Added additional commente.

With the current form of `ProcessLogs` we'll still need `IsCatchUpBlock`. At minimum, `ProcessLogs` would need to be modified to count all of the Warp messages to be processed for all of the catch-up blocks in order to fully populate the `keyManager` pending queues. If we do that, then the write validity would be fully encapsulated by the `DatabaseManager` and `keyManager`. I will explore making that change in this PR, since it should be equally valid here.

Reply:
@cam-schultz Not sure if you wrote this comment before reading through my feedback, but if we implement the min-heap for updating the most recent block, we shouldn't need to know if a block is a catchup block, since the height will just be added to the min-heap, and the non-catchup values will be processed out of the min-heap once all the catchup blocks have been processed.

Reply:
You're absolutely right. Requiring in-order writes turns out to be a powerful invariant. I've made this change now.